### PR TITLE
Add tool to generate metrics documentation

### PR DESCRIPTION
**Pull request by machadovilaca**
**What this PR does / why we need it**:

This PR adds a tool to automatically generate documentation with information about CDI exposed metrics.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: pkg/operator/controller/metrics.go [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+import (
+	promv1 "github.com/coreos/prometheus-operator/pkg/apis/monitoring/v1"
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promhttp"
+	logc "log"
+	"net/http"
+)
+
+// Handler creates a new prometheus handler to receive scrap requests
+func Handler(MaxRequestsInFlight int) http.Handler {
```
Comment (brybacki) [inline pkg/operator/controller/metrics.go:12]: It is more a of a MetricHandler or as a comment states MetricsScrapRequest Handler?

Now it looks like a main Handler for anything that `operator/controller/`. I think it is misleading and lacks context.

**Review by brybacki — CHANGES_REQUESTED** [review]
one comment regarding the `handler` name

File: pkg/operator/controller/metrics.go (hunk 1 above)
Comment (machadovilaca) [reply pkg/operator/controller/metrics.go]: updated

**Comment by machadovilaca** [comment]
@brybacki Just rebased, can you take a look again?

**Comment by brybacki** [comment]
quay had issues
/retest

**Comment by akalenyu** [comment]
@machadovilaca apparently recording rules can't have an annotations field:
https://github.com/kubevirt/hostpath-provisioner-operator/pull/195

**Comment by machadovilaca** [comment]
@akalenyu Moved the record rules information to a different structure (that also contains a description field). The record rules are created based on it but ignore the description, which is now only used to generate the metrics docs,

**Pull request merged**